conans/client/graph/install_graph.py

```diff
@@ -454,27 +454,62 @@ def install_build_order(self):
                   "order": [[n.serialize() for n in level] for level in install_order]}
         return result
 
-    def raise_errors(self):
+    def _get_missing_invalid_packages(self):
+        def analyze_package(package):
+            if package.binary == BINARY_MISSING:
+                missing.append(package)
+            elif package.binary == BINARY_INVALID:
+                invalid.append(package)
         missing, invalid = [], []
```

Comment:
```suggestion
        missing, invalid = [], []
        def analyze_package(package):
            if package.binary == BINARY_MISSING:
                missing.append(package)
            elif package.binary == BINARY_INVALID:
                invalid.append(package)
```
Even if the evaluation order of python allows it, I think the variables that are within a closure should at least have been defined before the function is defined, for readability.